cloudify_cli/tests/commands/test_events.py

```diff
@@ -31,34 +30,35 @@ def mock_log_message_prefix(event):
 
 class EventsTest(CliCommandTest):
 
-    events = []
-
     def setUp(self):
         super(EventsTest, self).setUp()
         self._create_cosmo_wd_settings()
         # execution will terminate after 10 seconds
-        self.execution_termination_time = time.time() + 10
+        self.execution_start_time = time.time()
+        self.execution_termination_time = self.execution_start_time + 10
+        self.events = self._generate_events(self.execution_start_time,
+                                            self.execution_termination_time)
         self.executions_status = Execution.STARTED
-        events_generator = threading.Thread(
-            target=self.generate_events,
-            args=(self.execution_termination_time,))
-        events_generator.daemon = True
-        events_generator.start()
-
-    def generate_events(self, execution_termination_time):
-        while time.time() < execution_termination_time:
-            # to simulate a common events flow, sleep for 3 secs every 2 secs
-            if len(EventsTest.events) > 0 and time.time() % 2 == 0:
```

Comment:
btw this didn't even really work: `time.time()` is a float and the chance of that %2 to equal exactly 0.0 is.. low :D